src/ios/OSBarcodeScanner.m

```diff
@@ -1,19 +1,34 @@
 #import "OSBarcodeScanner.h"
+#import "ScannerViewController.h"
 
 @implementation OSBarcodeScanner
 
+CDVInvokedUrlCommand* cdvCommand;
+
 - (void)scan:(CDVInvokedUrlCommand*)command
 {
-    CDVPluginResult* pluginResult = nil;
-    NSString* echo = [command.arguments objectAtIndex:0];
+    cdvCommand = command;
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                          selector:@selector(notificationHandler:)
+                                          name:@"barcode reader finished" object:nil];
+
+    UIViewController* scannerViewController = [[ScannerViewController alloc] initWithNibName:@"ScannerViewController" bundle:nil];
+    
+    if (@available(iOS 13.0, *)) {
+                [scannerViewController setModalPresentationStyle: UIModalPresentationFullScreen];
+            }
+    [self.viewController presentViewController:scannerViewController animated:true completion:nil];
+}
 
-    if (echo != nil && [echo length] > 0) {
-        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
+-(void)notificationHandler:(NSNotification *)notice{
+    NSString *str = [notice object];
+    CDVPluginResult* pluginResult = nil;
+    if (str != nil && [str length] > 0 && !([str isEqual: @"error"])) {
+        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:str];
     } else {
         pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
```

Comment:
Does it make sense to have a reason for the error?

Reply:
@vmfo22 on this I reviewed and yes. I changed to a a `CDVCommandStatus_NO_RESULT` it was a better response.